design-patterns/creational-patterns/src/objectPool/NotFreeInstanceException.java

```diff
@@ -0,0 +1,7 @@
+package objectPool;
+
+public class NotFreeInstanceException extends Exception {
+	public NotFreeInstanceException(){
```

Comment:
I believe a better name for that Exception would be 
```
NoAvailableIntstanceException()
```

What do you think ?

Reply:
sounds reasonable